src/status_im/extensions/camera.cljs

```diff
@@ -0,0 +1,67 @@
+(ns status-im.extensions.camera
+  (:require [re-frame.core :as re-frame]
+            [reagent.core :as reagent]
+            [status-im.utils.fx :as fx]
+            [status-im.ui.components.list-selection :as list-selection]
+            [status-im.qr-scanner.core :as qr-scanner]
+            [status-im.ui.screens.navigation :as navigation]
+            [status-im.utils.handlers :as handlers]))
+
+;; Common
+
+(handlers/register-handler-fx
+ :extensions/camera-cancel
+ (fn [_ [_ _ {{:keys [on-failure]} :data}]]
+   (when on-failure
+     (re-frame/dispatch (on-failure {:result "user cancelled"})))))
+
+(handlers/register-handler-fx
+ :extensions/camera-denied
+ (fn [_ [_ {{:keys [on-failure]} :data}]]
+   (when on-failure
+     (re-frame/dispatch (on-failure {:result "user denied access to camera"})))))
+
+;; Photo taker
+
+(handlers/register-handler-fx
+ :extensions/camera-error
+ (fn [cofx [_ error {:keys [on-failure]}]]
+   (fx/merge cofx
+             {:dispatch (on-failure {:result error})}
+             (navigation/navigate-back))))
+
+(handlers/register-handler-fx
+ :extensions/camera-picture-taken
+ (fn [cofx [_ base64 {{:keys [on-success]} :data}]]
+   (fx/merge cofx
+             {:dispatch (on-success {:result base64})}
+             (navigation/navigate-back))))
+
+(handlers/register-handler-fx
+ :extensions/camera-picture-selected
+ (fn [cofx [_ base64 {{:keys [on-success]} :data}]]
+   (navigation/navigate-back cofx)))
```

Comment:
why don't we call `on-success` here?

Reply:
the fx is not used anymore. I forgot removing it